src/core/components/ModelDropdownButton.tsx

```diff
@@ -0,0 +1,70 @@
+import styles from './NavBar.module.css';
+import { useCallback, useEffect, useState } from 'react';
+import { useAppContext } from '../context/useAppContext';
+
+interface ModelDropdownButtonProps {
+  dropdownOpen: boolean;
+  setDropdownOpen: (dropdownOpen: boolean) => void;
+}
+
+const ModelDropdownButton = ({
+  dropdownOpen,
+  setDropdownOpen,
+}: ModelDropdownButtonProps) => {
+  const [isDisabled, setIsDisabled] = useState(false);
+
+  const { modelConfig, messageHistoryStore } = useAppContext();
+
+  const ModelIconComponent = modelConfig.icon;
+
+  const toggleDropdown = useCallback(() => {
+    if (!isDisabled) {
+      setDropdownOpen(!dropdownOpen);
+    }
+  }, [dropdownOpen, isDisabled, setDropdownOpen]);
+
+  // Handle message history changes
+  const messages = messageHistoryStore.getSnapshot();
+  const hasUserMessages = messages.length > 1;
+
+  useEffect(() => {
+    if (hasUserMessages) {
+      setIsDisabled(true);
+      setDropdownOpen(false);
+    } else {
+      setIsDisabled(false);
+    }
+  }, [hasUserMessages, setDropdownOpen]);
```

Comment:
I don't think the user can actually enter a chat with the dropdown open, because as soon as they click outside the dropdown, it will close. But just to be safe let's be sure its closed.